packages/atlas/src/providers/currentBlock/currentBlock.manager.ts

```diff
@@ -0,0 +1,44 @@
+import React, { useEffect, useRef } from 'react'
+
+import { useJoystream } from '@/providers/joystream'
+
+import { useCurrentBlockStore } from './currentBlock.store'
+
+export const CurrentBlockManager: React.FC = () => {
+  const { setCurrentBlock, setCurrentBlockMsTimestamp } = useCurrentBlockStore((state) => state.actions)
+  const { joystream, proxyCallback } = useJoystream()
+  const firstRender = useRef(true)
+
+  // fetch current block from the chain, but only just once
+  useEffect(() => {
+    if (!firstRender.current || !joystream) {
+      return
+    }
+    joystream.getCurrentBlock().then((block) => {
+      setCurrentBlock(block)
```

Comment:
Let's also set the timestamp here, I guess I missed that in the initial implementation